### DISCUSSION TITLE
Semantic release workflow with branch per environment stratergy

### DISCUSSION THREAD

**Discussion by s1mrankaur** [semantic-release/semantic-release#2094] [Q&A]
In the CI-CD process, how(and when) does version bump happen? If directly merging through bitbucket/GitHub dashboard, then how to bump the version right before the merge? We won't do it after the merge because the CICD pipeline starts as soon as there is a change detected in the pipeline, and if a version bump was happening through the pipeline, it will just end up in a loop.

we have develop,  staging, and master branches. We do not wish to introduce new branches- just feels like a disruption and not so continuous. 

when and where should the version be bumped so that when the staging branch is merged into master- the version in the master is bumped as part of the same commit to avoid triggering the pipeline twice.

**Comment by travi** [comment]
the supported workflow configurations are described in our documentation: https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration
